src/HttpClient/HttpClientFactory.php

```diff
@@ -113,7 +113,9 @@ public function create(Options $options): HttpAsyncClientInterface
                 ]);
             }
 
-            throw new \RuntimeException('The "http_proxy" option requires either the "php-http/curl-client" or the "php-http/guzzle6-adapter" package to be installed.');
+            if (null === $this->httpClient) {
```

Comment:
What about making this the `else` statement instead of an additional `if`?

Reply:
Right. Don't know how I have missed that. Done, thanks.